azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Webhook.java

```diff
@@ -0,0 +1,651 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.containerregistry;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.containerregistry.implementation.WebhookInner;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.ExternalChildResource;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.Resource;
+import com.microsoft.azure.management.resources.fluentcore.model.Appliable;
+import com.microsoft.azure.management.resources.fluentcore.model.Attachable;
+import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+import com.microsoft.azure.management.resources.fluentcore.model.Refreshable;
+import com.microsoft.azure.management.resources.fluentcore.model.Settable;
+import com.microsoft.azure.management.resources.fluentcore.model.Updatable;
+import rx.Observable;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * An object that represents a webhook for a container registry.
+ */
+@Fluent
+@Beta(Beta.SinceVersion.V1_4_0)
+public interface Webhook extends
+    ExternalChildResource<Webhook, Registry>,
+    Resource,
+    HasInner<WebhookInner>,
+    Refreshable<Webhook>,
+    Updatable<Webhook.Update> {
+
+    /**
+     * @return the status of the webhook
+     */
+    boolean isEnabled();
+
+    /**
+     * @return the scope of repositories where the event can be triggered
+     * <p>
+     * For example:
+     *  - 'foo:*' means events for all tags under repository 'foo'
+     *  - 'foo:bar' means events for 'foo:bar' only
+     *  - 'foo' is equivalent to 'foo:latest'
+     *  - empty means all events
+     */
+    String scope();
+
+    /**
+     * @return the service URI for the webhook to post notifications
+     */
+    String serviceUri();
+
+    /**
+     * @return the Custom headers that will be added to the webhook notifications
+     */
+    Map<String, String> customHeaders();
+
+    /**
+     * @return the list of actions that trigger the webhook to post notifications
+     */
+    Collection<WebhookAction> triggers();
+
+    /**
+     * @return the provisioning state of the webhook
+     */
+    ProvisioningState provisioningState();
+
+    /**
+     * @return the webhook parent ID
+     */
+    String parentId();
+
+    /**
+     * Changes the status of the webhook to "enabled".
+     *
+     * @return the webhook
+     */
+    Webhook enable();
```

Comment:
why isn't this returning `void`? (same about `.disable()`)

Reply:
fixed